### DISCUSSION TITLE
Squash all revisions within a range

### DISCUSSION THREAD

**Discussion by Brixy** [jj-vcs/jj#4009] [Ideas]
Hi guys,

`squash --from --into` has been a real productivity boost. Thanks again.

I would like to suggest the following extension for `squash`:

```sh
jj squash -i --from rz..yq --into x
```
to squash `rx`, `yq` and all revisions 'between' them into `x`, where `-i` lets you choose the revisions interactively.

**My workflow:**

```
jj new -m 'featureXY'
```

Then I use `jj commit -m '..'` all the time and maybe `jj commit -m 'A sensible commit message'` in some cases.

Finally I would love to use something like the command suggested above to get a single `featureXY` revision.

(I know that `jj squash --from` now accepts multiple revisions: #3276.)

This would also be a great option to squash all revisions of an unnamed working branch into a single revision.

What do you think? Is something similar already possible?

If this makes sense, I am happy to open an issue.

Thanks a lot.

**Comment by ilyagr** [comment]
I *think* that `jj squash --from rz..yq --into rz` would already squash all the revisions between `rz` and `yq` into `rz`. You could also do `jj squash --from rz::yq --into x`. Is that right?

If so, then I think the main thing you are asking for is some sort of an interactive UI for choosing a subset of revisions to squash from. I think this could be useful, though I'd make a syntax something like:

```
jj squash --from ui:rz..yq --into x
```

(I initially thought of `interactive:rz..yq`, which is a little clearer, but we can keep thinking of the best prefix to use).

WDYT?

**Comment by Brixy** [comment]
Thank you very much!

Oh, I'm sorry. You are right. This is the mistake I made:

```sh
# This does not work and exits with "Nothing changed."
jj squash --from younger..older --into x

# This works; didn't test this
jj squash --from older..younger --into x

# --interactive works here, too
jj squash -i --from older..younger --into x
```

So, I just did not try to swap younger/older revisions. I just read the logs from top to bottom. Anyway, I'm very happy that this works. Squashing with `x..y` is a dream workflow.

Thank you very much for pointing this out to me.
